homeassistant/components/fritzbox/sensor.py

```diff
@@ -29,9 +34,58 @@ async def async_setup_entry(hass, config_entry, async_add_entities):
             entities.append(FritzBoxTempSensor(device, fritz))
             devices.add(device.ain)
 
+        if device.battery_level is not None:
+            entities.append(FritzBoxBatterySensor(device, fritz))
+            devices.add(f"{device.ain}_battery")
+
     async_add_entities(entities)
 
 
+class FritzBoxBatterySensor(SensorEntity):
+    """The entity class for Fritzbox battery sensors."""
+
+    def __init__(self, device, fritz):
+        """Initialize the sensor."""
+        self._device = device
+        self._fritz = fritz
+
+    @property
+    def device_info(self):
+        """Return device specific attributes."""
+        return {
+            "name": self.name,
+            "identifiers": {(FRITZBOX_DOMAIN, self._device.ain)},
+            "manufacturer": self._device.manufacturer,
+            "model": self._device.productname,
+            "sw_version": self._device.fw_version,
+        }
+
+    @property
+    def unique_id(self):
+        """Return the unique ID of the device."""
+        return f"{self._device.ain}_battery"
+
+    @property
+    def name(self):
+        """Return the name of the device."""
+        return f"{self._device.name} Battery"
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._device.battery_level
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement."""
+        return PERCENTAGE
+
```

Comment:
How is this updating?

Reply:
Because of the current implementation without DataUpdateCoordinator and the situation (prior to this PR) that per device only one entity was registered, each entity has it's own update method. In opposite to that, the `FritzBoxBatterySensor` is always an additional sensor of a device, which has a self updating entity, therefor device data are updated within that entity

Reply:
Looks like we fall though if there is no `update` method

https://github.com/home-assistant/core/blob/6e911ba19f9bb9b34cfd2981786540a6d485d4f9/homeassistant/helpers/entity.py#L497

So `async_write_ha_state` should still get called.

👍